### PR TITLE
test(coverage): raised unit test coverage from 47% to 80%

### PR DESCRIPTION
**Pull request by rios0rios0**
## Summary
- Refactored `FetchGitHubStats`, `FetchGitLabStats`, `FetchAzureDevOpsStats` to accept `*http.Client` parameter (nil = default) for testability
- Added `fetchers_test.go` with mock HTTP transport (`roundTripFunc`) tests covering all 3 platform fetchers and their language sub-functions
- Added tests for utility functions: `saveStatsHistory`, `loadTokenUsage`, `getEnvOrDefault`, `ColorScale`, `Generate*` wrappers, `languageColor`, `accumulateByYear` edge cases
- Covered error branches: HTTP failures, JSON parse errors, body read errors, non-200 status codes

Coverage: **47.2% → 80.1%** (948/1184 statements covered)

## Test plan
- [x] `make test` passes (all tests pass in <1s)
- [x] `go test -tags=unit -coverprofile=cover.out ./... && go tool cover -func=cover.out | tail -1` reports 80.1%
- [x] `go build -o /dev/null main.go` compiles

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: helpers_test.go [hunk 1]
```diff
@@ -872,6 +872,31 @@
...
+	})
+
+	t.Run("should handle unwritable README gracefully", func(t *testing.T) {
+		// given
+		dir := t.TempDir()
+		readmePath := filepath.Join(dir, "README.md")
+		content := "<details>\n\t<summary>2025</summary>\n\t<div align=\"center\">\n\t\t<img src=\"x\" />\n\t</div>\n</details>\n"
+		os.WriteFile(readmePath, []byte(content), 0644)
+		// Make file unwritable
+		os.Chmod(readmePath, 0444)
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply helpers_test.go:905]: Addressed in `e0aae1e`. Added error assertions for os.WriteFile and os.Chmod calls, with t.Skipf for platforms where chmod is not enforced.

File: helpers_test.go [hunk 2]
```diff
@@ -872,6 +872,31 @@
...
+
+		// when / then (should not panic, just log)
+		updateReadmeYearSections(readmePath, []int{2024, 2025}, "testuser")
+
+		// cleanup
+		os.Chmod(readmePath, 0644)
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply helpers_test.go:919]: Addressed in `e0aae1e`. The chmod calls now have error handling with t.Skipf/t.Logf fallbacks.

File: helpers_test.go [hunk 3]
```diff
@@ -872,6 +872,31 @@ func TestUpdateReadmeYearSections(t *testing.T) {
...
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply helpers_test.go]: Addressed in `e0aae1e`. Applied the suggestion — os.WriteFile now uses t.Fatalf, os.Chmod uses t.Skipf, and cleanup uses t.Logf.

File: helpers_test.go [hunk 4]
```diff
@@ -887,3 +912,411 @@ func TestUpdateReadmeYearSections(t *testing.T) {
...
+func TestSaveStatsHistoryError(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should return error when path is invalid", func(t *testing.T) {
+		// given
+		history := &StatsHistory{Version: 1}
+
+		// when
+		err := saveStatsHistory(history, "/nonexistent/dir/history.json")
+
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply helpers_test.go:1149]: Addressed in `e0aae1e`. Replaced with t.TempDir() plus a missing subdirectory path for deterministic failure.

File: helpers_test.go [hunk 5]
```diff
@@ -887,3 +912,411 @@ func TestUpdateReadmeYearSections(t *testing.T) {
...
+		require.NoError(t, err)
+		require.Len(t, result, 2)
+		assert.Equal(t, "2026-03-15", result[0].Date)
+		assert.Equal(t, "2026-03-16", result[1].Date)
+	})
+
+	t.Run("should return error when file does not exist", func(t *testing.T) {
+		// when
+		result, err := loadTokenUsage("/nonexistent/path.json")
+
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply helpers_test.go:1186]: Addressed in `e0aae1e`. Now uses t.TempDir() with a nonexistent filename for deterministic NotExist error.

File: fetchers_test.go [hunk 6]
```diff
@@ -0,0 +1,1468 @@
...
+			}
+			if strings.Contains(path, "/commits") && langPhase && strings.Contains(query, "$top=1") {
+				return jsonResponse(http.StatusOK, latestCommit), nil
+			}
+			if strings.Contains(path, "/commits") && langPhase {
+				return jsonResponse(http.StatusOK, allCommits), nil
+			}
+			if strings.Contains(path, "/commits/") {
+				return jsonResponse(http.StatusOK, commitDetail), nil
+			}
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply fetchers_test.go]: Addressed in `e0aae1e`. Reordered conditionals so `/commits/{id}` (commit detail) is matched before `/commits` (commit list), preventing the language-fetch path from being short-circuited.
